### ISSUE TITLE
Shopkeeper addmember / removemember

### ISSUE THREAD

**Issue by blablubbabcDEV** [Shopkeepers/Shopkeepers#342]
**Migrated from:** https://dev.bukkit.org/projects/shopkeepers/issues/341

**Originally posted by LeB1ack (Sep 15, 2015):** 

What about a command to add/remove Members to a Shopkeeper.The registered member should be able to edit the offers and access the chest.

**Comment by blablubbabcDEV** [comment]
**Originally commented by FatherWh0 (Sep 7, 2016):** 

We have a mall with shops rentable via the plugin Areashops. AS has an addfriend argument which can be used to add a member to the rented region. This has allowed some of our players to partner up. A few of the players have requested the option to allow their partners to restock their chests. This would not require them to edit trades but only to access chest contents. Such an added feature would be an enhancement to our shopkeepers experience.

**Comment by blablubbabc** [comment]
Possible implementation that avoids commands for player shops: 3 editor buttons for adding, removing and listing shop owners/members. Memberss get specified in chat, similar to how naming works.
Or even nicer (but might not be possible..): Editor button that opens up a book where the player edits the list of shop members.

Potential problem: Which rights do members have?
* Members=Owners
* Access chest
* Edit trades
* Edit shop object
* Edit memberships (add/remove members, add/remove owners)

There might be usecases for either, on a per-shop basis, but implementing all these options on a per-shop basis might make the plugin usage complicated...
One option may be to make (some of) these server-wide configurable in the config.
The easiest to implement would be members=owners or members can only access the chest.

**Comment by KironDevCoder** [comment]
I'd say members should be able to access chest and edit trades as default, but it being configurable in the server config.

**Comment by impmallet** [comment]
I have many users asking for this feature. I understand that it is certainly not a priority, but hope that you would still continue adding it to your to do list. I agree with KironDevCoder on permissions aspect of the proposed feature.

Also, as mentioend above, a trust player button inside of the shopkeeper menu would like be the most painless option. It would be nice to also have commands for admins to use to add/remove members from admin shopkeepers or allow the transfer of ownership of a shopkeeper from one player to another (in case 2 people are playing together and one decides to quit).

Thank you!

**Comment by blablubbabc** [comment]
@impmallet
> It would be nice to also have commands for admins to use to add/remove members from admin shopkeepers

Admin shops are not owned by anyone, so I am not completely sure what you suggest there.

> allow the transfer of ownership of a shopkeeper from one player to another

There is already a transfer command.

**Comment by impmallet** [comment]
Thank you for your response.

Are you ever planning on adding the original feature requested in this ticket?

**Comment by blablubbabc** [comment]
> Are you ever planning on adding the original feature requested in this ticket?

Yes, I just don't have much time and usually more important things to work on.

**Comment by JaHollTV** [comment]
Hey there,

could you please continue this Idea?

Would be very, very useful for RPG-Servers, where Players manage a Shop together.
I think Shoopkeeper is primarily used on RPG-Servers, so this feature would fit in 100%

**Comment by 2000erikandersson-maker** [comment]
Hey! Amazing work, I assume you still have alot to do and this is low on the list but I was wondering now in 2026 if there is any update to it? Is it possible to have a player resupply otherways like a hopper where only "locked (selling items)" is able to refill?

Regars Xome

**Comment by lerndmina** [comment]
I opened a PR to add this feature here as I wanted it on my own server. The PR was vibe coded but has been tested by me and the server I run. I don't expect it to be merged but it might be useful to you, it was useful to me.

**Comment by blablubbabc** [comment]
This is available now in v2.27.0
